LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIStrategyMapItem_AlienNetworkComponent_LW.uc

```diff
@@ -0,0 +1,347 @@
+
+	super(UIStrategyMapItem).UpdateFromGeoscapeEntity(GeoscapeEntity);
+
+	InfiltratingSquad = GetInfiltratingSquad();
+	 
+	MissionState = GetMission();
+	if(InfiltratingSquad != none)
+	{
+		InfiltrationPct = int(InfiltratingSquad.CurrentInfiltration * 100.0);
+		InfiltrationPctValue = string(InfiltrationPct) $ "%";
+		InfilPct.SetHTMLText(CenterText(class'UIUtilities_Text'.static.AddFontInfo(InfiltrationPctValue, false, true,, 20)));
+
+		if (!bScanButtonResized)
+		{
+			MissionTitle = class'UIUtilities_Text'.static.CapsCheckForGermanScharfesS(InfiltratingSquad.sSquadName);
+			MissionTemplate = class'X2MissionTemplateManager'.static.GetMissionTemplateManager().FindMissionTemplate(MissionState.GeneratedMission.Mission.MissionName);
+			MissionInfo = class'UIUtilities_Text'.static.CapsCheckForGermanScharfesS(MissionTemplate.PostMissionType); 
+
+			ScanButton.SetText(MissionTitle, MissionInfo, " ", " "); // have to leave blank spaces so that Flash will size BG big enough 
+			SetLevel(0); // show 0 doom pips
+
+			InfilLabel.SetHTMLText(CenterText(class'UIUtilities_Text'.static.GetSizedText(strRecon, 16)));
+			bScanButtonResized = true;
+
+			CachedImageName = InfiltratingSquad.GetSquadImagePath();
+			SquadImage.LoadImage(CachedImageName);
+		}
+		ScanButton.DefaultState();
+		ScanButton.PulseScanner(false);
+		ScanButton.ShowScanIcon(false);
+		ScanButton.Realize();
+
+		// but the MC queuing system means that the width isn't set here the first time it's been invoked
+		ScanWidth = ScanButton.MC.GetNum("bg._width"); 
+
+		if (ScanWidth != CachedScanButtonWidth)
+		{
+			InfilPct.SetX(ScanWidth - 60);
+			InfilLabel.SetX(ScanWidth - 65);
+			ScanButton.SetX(- (ScanWidth/2));
+			
+			CachedScanButtonWidth = ScanWidth;
+
+			SquadImage.SetPosition(-(ScanWidth/2 + 48), 0);
+		}
+		if (CachedImageName != InfiltratingSquad.GetSquadImagePath())
+		{
+			CachedImageName = InfiltratingSquad.GetSquadImagePath();
+			SquadImage.LoadImage(CachedImageName);
+		}
+
+		ProgressBar.Hide();
+	}
+	else if (MissionState.MakesDoom())
+	{
+		SetLevel(MissionState.Doom);
+		ProgressBar.Hide();
+		ScanButton.Hide();
+		bScanButtonResized = false;
+	}
+	else
+	{
+		ProgressBar.Show();
+		UpdateProgressBar(MissionState);
+
+		ScanButton.Hide();
+		SquadImage.Hide();
+		bScanButtonResized = false;
+	}
+}
+
+simulated function string CenterText(string Text)
+{
+	return class'UIUtilities_Text'.static.AlignCenter(Text);
+}
+
+simulated function XComGameState_LWPersistentSquad GetInfiltratingSquad()
+{
+	return `LWSQUADMGR.GetSquadOnMission(GeoscapeEntityRef);
+}
+
+simulated function XComGameState_MissionSite GetMission()
+{
+	return XComGameState_MissionSite(`XCOMHISTORY.GetGameStateForObjectID(GeoscapeEntityRef.ObjectID));
+}
+
+function OpenInfiltrationMissionScreen()
+{
+	local UIMission_LWLaunchDelayedMission MissionScreen;
+	local XComHQPresentationLayer HQPres;
+	
+	// WOTC DEBUGGING
+	`LWTrace(">> Launching delayed mission UI");
+	// END

```

Comment:
I would remove this if it's no longer needed for debugging.

Reply:
done